lib/Default/SmrShip.class.inc

```diff
@@ -120,11 +120,15 @@ class SmrShip extends AbstractSmrShip {
 	}
 	
 	function updateHardware() {
-		if($this->hasChangedHardware === true) {
+		if (!empty($this->hasChangedHardware)) {
 			$this->db->lockTable('ship_has_hardware');
```

Comment:
This can probably be removed, with the sector lock in place this should be redundant, and beyond that if the value has already been changed in the background it's not like we refetch it here, so the damage would already have been done and this wouldn't stop it but would likely slow down a load of things

Reply:
Ah yeah, I suppose the sector lock supersedes this hardware lock. But in that case, the only cost incurred by this lock is the expense of `lockTable` and `unlock`, right?

Actually, a more worrying thought just occurred to me. `MySqlDatabase::unlock` unlocks _all_ tables. Does that mean that `updateHardware` (and all other places that call `unlock`) unlocks the sector lock as well?

Reply:
They don't touch the sector lock (it's a table itself and doesn't use the mysql lock commands), but the cost of the lock/unlock also includes the cost that no other hardware updates can happen at the same time eg someone uno'ing on one side of the universe will be blocked by someone hitting mines on the other side of the universe